src/ApiCompat/Rules/Compat/TypeCannotChangeClassification.cs

```diff
@@ -20,8 +21,19 @@ public override DifferenceType Diff(IDifferences differences, ITypeDefinition im
             if (implObjType != contractObjType)
             {
                 differences.AddIncompatibleDifference(this,
-                    "Type '{0}' is a {1} in the implementation but is a {2} in the contract.",
+                    "Type '{0}' is a '{1}' in the implementation but is a '{2}' in the contract.",
                     impl.FullName(), implObjType, contractObjType);
+
+                return DifferenceType.Changed;
+            }
+
+            if (impl.Attributes.HasIsReadOnlyAttribute() && !contract.Attributes.HasIsReadOnlyAttribute())
+            {
+                differences.AddIncompatibleDifference(this,
+                    "Type '{0}' is marked as readonly in the implementation so it must also be marked readonly in the contract.",
```

Comment:
I think this should be other way around: 
- readonly in implementation only is ok
- readonly in contract only is not ok

Reply:
Fixed.